R/types.R

```diff
@@ -685,7 +685,7 @@ is_copyable <- function(x) {
 is_equal <- function(x, y) {
   identical(x, y)
 }
-#' @useDynLib rlang rlang_is_identical
-is_identical <- function(x, y) {
-  .Call(rlang_is_identical, x, y)
+#' @useDynLib rlang rlang_is_reference
+is_reference <- function(x, y) {
```

Comment:
`is_aliased(x)` could return `NAMED(x) > 2`?  (although that would require some gymnastics to make sure we don't take a reference inside the function)